packages/web-components/src/components/masthead/masthead-composite.ts

```diff
@@ -699,15 +699,15 @@ class DDSMastheadComposite extends LitElement {
         ${!hasSearch
           ? undefined
           : html`
-              <dds-masthead-search-composite
+              <dds-masthead-search-container
```

Comment:
Composite component should never depend on container components. Do you have any background on this change? In general, container component should connect backend API to composite components.

Reply:
The reason behind this change is that the Masthead with Custom Data story is currently using the masthead composite directly in order to bypass the retrieved navLinks from the Translations API. However, this has the side effect of not loading the SearchAPI actions that usually come from masthead container. By using search-container over search-composite we are able to have autosuggestions when using masthead-composite. 

I've looked into adding the Search API into masthead-composite itself, though not sure if that's possible -- please let me know otherwise.

Reply:
For that, you can create a new container component that works only with search results. You can start with creating a copy of `masthead-container.ts`. In the new copy (make sure assigning a new class name as well as a new custom element tag name), `mapStateToProps()` should pick only [`currentSearchResults`](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.16.1/packages/web-components/src/components/masthead/masthead-container.ts#L101) and `mapDispatchToProps()` should pick only [`_loadSearchResults`](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/v1.16.1/packages/web-components/src/components/masthead/masthead-container.ts#L118). In this way, the only API the new container component interacts with will be `SearchAPI`.

Reply:
I see, that makes sense, it's essentially a "variation" of the container. I was focused on finding a solution between these two components that I hadn't considered creating a third. Thanks for letting me know!